src/Microsoft.AspNetCore.Razor/CodeGenerators/CSharpCodeGenerator.cs

```diff
@@ -82,12 +83,32 @@ public override CodeGeneratorResult Generate()
                             csharpCodeVisitor.Accept(Tree.Children);
                         }
                     }
+
+                    BuildAfterExecuteContent(writer, Tree.Children);
                 }
             }
 
             return new CodeGeneratorResult(writer.GenerateCode(), writer.LineMappingManager.Mappings);
         }
 
+        /// <summary>
+        /// Provides an entry point to append code (after execute content) to a generated Razor class.
+        /// </summary>
+        /// <param name="writer">The <see cref="CSharpCodeWriter"/> to receive the additional content.</param>
+        /// <param name="chunks">The list of <see cref="Chunk"/>s for the generated program.</param>
+        protected virtual void BuildAfterExecuteContent(CSharpCodeWriter writer, IList<Chunk> chunks)
+        {
+        }
+
+        /// <summary>
+        /// Provides an entry point to modify chunk data before code generation.
+        /// </summary>
+        /// <param name="context">The <see cref="CodeGeneratorContext"/> of the chunks."/> </param>
+        /// <param name="chunks">The list of <see cref="Chunk"/>s for the generated program.</param>
+        protected virtual void DecorateChunks(CodeGeneratorContext context, IList<Chunk> chunks)
```

Comment:
So we don't technically need this piece. MVC already semi-decorates the chunk tree by merging in _ViewImports chunks.

See: https://github.com/aspnet/Mvc/blob/c942eab6e268714f23f0d3b31e35036901d3b2e0/src/Microsoft.AspNetCore.Mvc.Razor.Host/MvcRazorHost.cs#L328-L331